indico/modules/users/export.py

```diff
@@ -0,0 +1,320 @@
+    """Return a deduplicated list of editable files.
+
+    When a file is not changed between revisions, multiple revisions might refer to
+    the same physical file. We don't want to include duplicate files in the export.
+    """
+    files = {file.file_id: file for revision in editable.revisions for file in revision.files}
+    return files.values()
+
+
+def build_storage_path(file):
+    """Build a path under which a given file be stored in the exported zip file.
+
+    The path includes both the id (to ensure uniqueness) and the title/name for
+    easy navigation.
+    """
+    if isinstance(file, RegistrationData):
+        event = file.registration.event
+        prefix = 'registrations'
+        path = f'{event.id}_{event.title}'
+    elif isinstance(file, AttachmentFile):
+        prefix = 'attachments'
+        path = ''
+    elif isinstance(file, AbstractFile):
+        event = file.abstract.event
+        prefix = 'abstracts'
+        path = f'{event.id}_{event.title}/{file.abstract.id}_{file.abstract.title}'
+    elif isinstance(file, PaperFile):
+        event = file._contribution.event
+        prefix = 'papers'
+        path = f'{event.id}_{event.title}/{file._contribution.id}_{file.paper.title}'
+    else:
+        editable = file.revision.editable
+        event = editable.contribution.event
+        prefix = f'editables/{editable.type.name}'
+        path = f'{event.id}_{event.title}/{editable.id}_{editable.contribution.title}'
+
+    path = secure_path(path)
+    filename = build_filename(file)
+    return str(Path() / prefix / path / filename)
+
+
+def build_filename(file):
+    file = getattr(file, 'file', file)
+    if isinstance(file, RegistrationData):
+        id = f'{file.registration_id}_{file.field_data_id}'
+    else:
+        id = file.id
+
+    filename = secure_filename(Path(file.filename).stem, '')
+    return f'{id}_{filename}.{file.extension}'
+
+
+def secure_path(path):
+    parts = (secure_filename(p, uuid4()) for p in Path(path).parts)
+    return Path(*parts)
+
+
+def convert_to_yaml(data):
+    """Convert data to yaml with a nicer indentation style.
+
+    Normal output does not indent lists, e.g.:
+        key:
+        - 1
+        - 2
+
+    Using this function, the list becomes indented:
+        key:
+          - 1
+          - 2
+
+    See: https://github.com/yaml/pyyaml/issues/234
+    """
+    class Dumper(yaml.Dumper):
+        def increase_indent(self, flow=False, indentless=False):
+            return super().increase_indent(flow, indentless=False)
+
+    return yaml.dump(data, Dumper=Dumper, allow_unicode=True)
+
+
+def get_old_requests(days):
+    """Get successful export requests older than the specified number of days."""
+    return (DataExportRequest.query
+            .filter(DataExportRequest.requested_dt < (now_utc() - timedelta(days=days)),
+                    DataExportRequest.state == DataExportRequestState.success)
+            .all())
+
+
+@email_sender
```

Comment:
Ugh I really need to remove this stupid decorator altogether some day. It's one of those things that someone added after they took the 'advanced python' course and thought decorators are the hail mary :D

IMHO it's much clearer to have `send_email(make_email(...))` where you immediately see "ok that sends an email"...